include/common/buf.h

```diff
@@ -8,16 +8,28 @@
 #ifndef LABWC_BUF_H
 #define LABWC_BUF_H
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
-
 struct buf {
+	/**
+	 * Pointer to underlying string buffer. If alloc != 0, then
+	 * this was allocated via malloc().
+	 */
```

Comment:
I know this is "original" - but we could consider `s/buf/data/` here. Seems consistent with other string implementations, including that of glib. Also avoid `buf->buf` which looks a big ugly.

Reply:
Renamed to `data`.